rm/rm.c

```diff
@@ -2618,33 +2693,31 @@ M0_INTERNAL int m0_rm_owner_loan_debit(struct m0_rm_owner *owner,
 
 	M0_PRE(owner != NULL);
 	M0_PRE(paid_loan != NULL);
-	M0_ENTRY("credit: %llu",
+	M0_ENTRY("loan=%p credit: %llu", paid_loan,
 			(long long unsigned) paid_loan->rl_credit.cr_datum);
 
 	m0_rm_ur_tlist_init(&retain_list);
 	m0_rm_ur_tlist_init(&remove_list);
 	m0_tl_for (m0_rm_ur, list, cr) {
 		if (!cr->cr_ops->cro_intersects(&paid_loan->rl_credit, cr))
-			m0_rm_ur_tlist_move(&retain_list, cr);
-		else {
-			loan = bob_of(cr, struct m0_rm_loan,
-				      rl_credit, &loan_bob);
-			if (!m0_cookie_is_eq(&loan->rl_cookie,
-					     &paid_loan->rl_cookie)) {
-				m0_rm_ur_tlist_move(&retain_list, cr);
-				continue;
-			}
-			rc = remnant_loan_get(loan, &paid_loan->rl_credit,
-					      &remnant_loan);
-			if (rc == 0) {
-				if (credit_is_empty(&remnant_loan->rl_credit)) {
-					m0_rm_ur_tlist_move(&remove_list, cr);
-					m0_rm_ur_tlist_add(&remove_list,
-						&remnant_loan->rl_credit);
-				} else
-					m0_rm_ur_tlist_move(&retain_list, cr);
-			}
-		}
+			continue;
+
+		loan = bob_of(cr, struct m0_rm_loan, rl_credit, &loan_bob);
+
+		if (!m0_cookie_is_eq(&loan->rl_cookie, &paid_loan->rl_cookie))
+			continue;
```

Comment:
and here, the "cr" will not be moved to tail.

Reply:
Yes, because we didn't move it out in the 1st place. There is no need to move cr from and back in if it not intersects with the debited credit, is it?

Reply:
Please note: this change is not part of the fix, it goes in a separate commit - https://github.com/Seagate/cortx-motr/pull/304/commits/a720ed38d169bbeb16e171f9b999b471b1ca071d. And it's desirable that it's landed in the separate commit also.

Reply:
agree.